webapp/src/test/java/com/box/l10n/mojito/rest/textunit/StringToDateTimeConverterTest.java

```diff
@@ -12,62 +13,69 @@ public class StringToDateTimeConverterTest {
+    ZonedDateTime result = instance.convert(source);
+    assertEquals(expResult.toInstant(), result.toInstant());
   }
 
   @Test
   public void testConvertISO() {
     String source = "2018-05-09T17:34:55.000Z";
     StringToDateTimeConverter instance = new StringToDateTimeConverter();
-    DateTime expResult = new DateTime("2018-05-09T17:34:55.000Z");
-    DateTime result = instance.convert(source);
-    assertEquals(expResult, result);
+    ZonedDateTime expResult = ZonedDateTime.parse("2018-05-09T17:34:55.000Z");
+    ZonedDateTime result = instance.convert(source);
+    assertEquals(expResult.toInstant(), result.toInstant());
   }
 
   @Test
   public void testConvertISONoTZ() {
     String source = "2018-05-09T17:34:55.000";
     StringToDateTimeConverter instance = new StringToDateTimeConverter();
-    DateTime expResult = new DateTime("2018-05-09T17:34:55.000Z");
-    DateTime result = instance.convert(source);
-    assertEquals(expResult, result);
+    ZonedDateTime expResult = ZonedDateTime.parse("2018-05-09T17:34:55.000Z");
+    ZonedDateTime result = instance.convert(source);
+    assertEquals(expResult.toInstant(), result.toInstant());
   }
 
   @Test
   public void testConvertISOUTC() {
-    String source = "2018-05-09T17:34:55.000";
+    String source = "2018-05-09T17:34:55.000Z";
     StringToDateTimeConverter instance = new StringToDateTimeConverter();
-    DateTime expResult = new DateTime("2018-05-09T17:34:55.000Z");
-    DateTime result = instance.convert(source);
-    assertEquals(expResult, result);
+    ZonedDateTime expResult = ZonedDateTime.parse("2018-05-09T17:34:55.000Z");
+    ZonedDateTime result = instance.convert(source);
+    assertEquals(expResult.toInstant(), result.toInstant());
   }
 
   @Test
   public void testConvertISOTZ() {
     String source = "2018-05-09T17:34:55.000-07:00";
     StringToDateTimeConverter instance = new StringToDateTimeConverter();
-    DateTime expResult = new DateTime("2018-05-10T00:34:55.000Z");
-    DateTime result = instance.convert(source);
-    assertEquals(expResult, result);
+    ZonedDateTime expResult = ZonedDateTime.parse("2018-05-10T00:34:55.000Z");
+    ZonedDateTime result = instance.convert(source);
+    assertEquals(expResult.toInstant(), result.toInstant());
   }
 
   @Test
   public void testConvertISOTZNoMillisecond() {
     String source = "2018-05-09T17:34:55-07:00";
     StringToDateTimeConverter instance = new StringToDateTimeConverter();
-    DateTime expResult = new DateTime("2018-05-10T00:34:55.000Z");
-    DateTime result = instance.convert(source);
-    assertEquals(expResult, result);
+    ZonedDateTime expResult = ZonedDateTime.parse("2018-05-10T00:34:55.000Z");
+    ZonedDateTime result = instance.convert(source);
+    assertEquals(expResult.toInstant(), result.toInstant());
+  }
+
+  @Test(expected = DateTimeParseException.class)
+  public void testConvertInvalidDateFormat() {
+    String source = "09-05-2018T17:34:55-07:00";
```

Comment:
is that a format that was encountered while testing?

Reply:
No, this format shouldn't be parsed because does not follow the expected ISO format. We were missing test cases that throw exceptions, and I added this one.